pymove/utils/integration.py

```diff
@@ -0,0 +1,970 @@
+
+            # calculing distances to idx
+            distances = np.float64(
+                haversine(
+                    lat_user,
+                    lon_user,
+                    df_pois[LATITUDE].to_numpy(dtype=np.float64),
+                    df_pois[LONGITUDE].to_numpy(dtype=np.float64),
+                )
+            )
+
+            # get index to arg_min and min distance
+            index_min = np.argmin(distances)
+            current_distances[idx] = np.min(distances)
+
+            # setting data for a single object movement
+            ids_POIs[idx] = df_pois.at[index_min, label_id]
+            tag_POIs[idx] = df_pois.at[index_min, label_poi]
+
+        df_[ID_POI] = ids_POIs
+        df_[DIST_POI] = current_distances
+        df_[TYPE_POI] = tag_POIs
+
+        print('Integration with POI was finalized')
+        print(
+            '\nTotal Time: {:.2f} seconds'.format((time.time() - start_time))
+        )
+
+    except Exception as e:
+        print('id: {}\n'.format(idx))
+        raise e
+
+
+
+def join_with_pois_optimizer(
+    df_,
+    df_pois,
+    label_poi='POI',
+    label_poi_name='name',
+    dist_poi=None,
+    reset_index=True,
+):
+
+    """
+    Performs the integration between trajectories and points
+    of interest, generating two new columns referring to the
+    name and distance from the nearest point of interest,
+    within the limit of distance determined by the parameter 'dist_poi',
+    of each point in the trajectory.
+
+    Parameters
+    ----------
+    df_ : dataframe
+        The input trajectory data.
+
+    df_pois : dataframe
+        The input point of interest data.
+
+    label_poi : String, optional("POI" by default)
+        Label of df_pois referring to the Point of Interest name.
+
+    dist_poi : List
+        List containing the distance limit to classify the most nearest point of
+        interest of each trajectory point.
+
+    reset_index : Boolean, optional(True by default)
+        Flag for reset index of the df_pois and df_ dataframes before the join.
+
+    Returns
+    -------
+    """
+
+    if dist_poi is None:
+        dist_poi = [10]
+    try:
+        print('Integration with POIs...')
+        start_time = time.time()
+
+        if df_pois[label_poi].unique() == len(dist_poi):
+            # get a vector with windows time to each point
+            if reset_index:
+                print('... reseting and dropping index')
+                df_.reset_index(drop=True, inplace=True)
+                df_pois.reset_index(drop=True, inplace=True)
+
+            # create numpy array to store new colum to dataframe of movement objects
+            ids_POIs = np.full(df_.shape[0], np.NAN, dtype='object_')
+            tag_POIs = np.full(df_.shape[0], np.NAN, dtype='object_')
+
+            minimum_distances = np.full(
+                df_.shape[0], np.Infinity, dtype=np.float64
+            )
+            shape_POIs = df_pois.shape[0]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'shape_POIs' is assigned to but never used (F841)](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5610065)